db/keyval/redis/plugin_impl_redis.go

```diff
@@ -57,6 +64,28 @@ func (p *Plugin) Init() error {
 	return p.Skeleton.Init()
 }
 
+// AfterInit is called by the Agent Core after all plugins have been initialized.
+func (p *Plugin) AfterInit() error {
+	if p.disabled {
+		return nil
+	}
+
+	// Register for providing status reports (polling mode)
+	if p.StatusCheck != nil {
+		p.StatusCheck.Register(core.PluginName(p.String()), func() (statuscheck.PluginState, error) {
+			_, _, err := p.Skeleton.NewBroker("/").GetValue(healthCheckProbeKey, nil)
```

Comment:
If possbile try to use directly connection broker to not create new instance again